cli/repl.rs

```diff
@@ -1,73 +1,240 @@
 // Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
 
-use crate::deno_dir::DenoDir;
+use crate::global_state::GlobalState;
+use crate::inspector::InspectorSession;
 use deno_core::error::AnyError;
+use deno_core::serde_json::json;
+use rustyline::error::ReadlineError;
+use rustyline::validate::MatchingBracketValidator;
+use rustyline::validate::ValidationContext;
+use rustyline::validate::ValidationResult;
+use rustyline::validate::Validator;
 use rustyline::Editor;
-use std::fs;
-use std::path::PathBuf;
+use rustyline_derive::{Completer, Helper, Highlighter, Hinter};
+use std::sync::Arc;
+use std::sync::Mutex;
 
-pub struct Repl {
-  editor: Editor<()>,
-  history_file: PathBuf,
+#[derive(Completer, Helper, Highlighter, Hinter)]
```

Comment:
Does using `derive` here actually provide completions/hints/highlights? Or is it just to have default impls?

Reply:
Just to have default no-op implementations because the `Helper` trait requires them all to be implemented.